python/ray/test/test_queue.py

```diff
@@ -10,9 +10,18 @@
 from ray.experimental.queue import Queue, Empty, Full
 
 
-def setup_module():
-    if not ray.worker.global_worker.connected:
-        ray.init()
+@pytest.fixture
+def ray_start():
+    # Start the Ray process.
+    ray.init()
+    yield None
+    # The code after the yield will run as teardown code.
+    ray.shutdown()
+
+
+def teardown_module():
```

Comment:
This should be unnecessary if we are passing `ray_start` in everywhere, right?

Reply:
Oops sorry I thought I took out `ray_start` since it did not work for `test_async` for some reason

Reply:
Does it actually not work with `test_async`? Any idea why that is?

Reply:
I'm not really sure, it just hangs on when we try to get the results `result = set(ray.get(consumers))` from the `Queue` https://github.com/ray-project/ray/pull/4012/files#diff-a97079c8e5770f39616578c37673e242R56